File: taloa-api/app/api/v1/routes/care.py
"""Pet Sitter Share (Etapa 25) — rotas de care links.

CRUD do dono (auth) + leitura publica por token (expira/revoga server-side).
"""
from fastapi import APIRouter, Depends, status

from app.core.security import get_current_user
from app.schemas.auth import CurrentUser
from app.schemas.care import CareProfile, CareShare, CareShareCreate
from app.services import care_service

router = APIRouter(tags=["care"])


@router.post("/care-shares", response_model=CareShare, status_code=status.HTTP_201_CREATED)
def create_care_share(
    body: CareShareCreate, user: CurrentUser = Depends(get_current_user)
) -> CareShare:
    return care_service.create_share(user, body)


@router.get("/care-shares/pet/{pet_id}", response_model=list[CareShare])
def list_care_shares(
    pet_id: str, user: CurrentUser = Depends(get_current_user)
) -> list[CareShare]:
    return care_service.list_shares(user, pet_id)


@router.delete("/care-shares/{share_id}")
def revoke_care_share(
    share_id: str, user: CurrentUser = Depends(get_current_user)
) -> dict:
    care_service.revoke_share(user, share_id)
    return {"status": "revoked"}


@router.get("/care/{token}", response_model=CareProfile)
def get_care(token: str) -> CareProfile:
    """Publico: perfil completo do pet se o token for valido e nao expirado."""
    return care_service.get_care(token)

File: taloa-web/lib/api/care.ts
// Pet Sitter Share — CRUD autenticado (dono) + leitura publica por token.
import { apiFetch } from "@/lib/api/client";
import type { CareDuration, CareProfile, CareShare } from "@/types/care";

const API_URL = process.env.NEXT_PUBLIC_API_URL ?? "http://localhost:8000";

export function createCareShare(
  petId: string,
  duration: CareDuration,
): Promise<CareShare> {
  return apiFetch<CareShare>("/v1/care-shares", {
    method: "POST",
    body: JSON.stringify({ pet_id: petId, duration }),
  });
}

export function listCareShares(petId: string): Promise<CareShare[]> {
  return apiFetch<CareShare[]>(`/v1/care-shares/pet/${petId}`);
}

export function revokeCareShare(id: string): Promise<unknown> {
  return apiFetch(`/v1/care-shares/${id}`, { method: "DELETE" });
}

// Publico (carer): perfil completo se o token for valido/nao expirado.
export async function getCare(token: string): Promise<CareProfile | null> {
  const res = await fetch(`${API_URL}/v1/care/${token}`, { cache: "no-store" });
  if (!res.ok) return null;
  return res.json();
}
